### PR TITLE
Adds Embedded Sources and enable deterministic builds for Core module

### LINKED ISSUE #1655: Fix NuGet health of the latest Nupkg
Fix the health of the latest [0.8.44-rc nupkg](https://nuget.info/packages/Microsoft.DataApiBuilder/0.8.44-rc)

image

Health of [0.7.6 nupkg](https://nuget.info/packages/Microsoft.DataApiBuilder/0.7.6):

image

### PR DESCRIPTION
**Pull request by severussundar**
## Why make this change?

-  Closes #1655 
  - Health of [0.8.44-rc nupkg](https://nuget.info/packages/Microsoft.DataApiBuilder/0.8.44-rc) nupkg needs to be fixed

## What is this change?

- To fix the NuGet generated in the `main` branch, `Product` and `Core` modules needs to be updated. `Product` module was introduced later and `0.8.*` NuGets do not have the Product module. So, the fix is broken into two PRs. 
- This PR adds changes only to the `Core` module. After merging to `main`, this PR can be ported over to `release/0.8` branch.  
- [PR #1657 ](https://github.com/Azure/data-api-builder/pull/1657) adds the changes to `Product` module.
 
- Sets `EmbedUntrackedSources` and `ContinuousIntegrationBuild` to `true`

## How was this tested?

- [x] Manual Tests
By applying the change on top of the `release/0.8` branch, the health of the NuGet generated was validated 
image

### CODE REVIEW

File: src/Core/Azure.DataApiBuilder.Core.csproj [hunk 1]
```diff
@@ -4,6 +4,7 @@
     <TargetFramework>net6.0</TargetFramework>
     <ImplicitUsings>enable</ImplicitUsings>
     <Nullable>enable</Nullable>
+    <EmbedUntrackedSources>true</EmbedUntrackedSources>
```
Comment (seantleonard) [inline src/Core/Azure.DataApiBuilder.Core.csproj:7]: Docs state for this setting:
> Set EmbedUntrackedSources to true to instruct the build system to embed project source files that are not tracked by the source control or imported from a source package to the generated PDB.

by " project source files that are not tracked by the source control " do they mean the actual build executable?

**Review by seantleonard — APPROVED** [review]
thanks for adding. one question.

**Review by Aniruddh25 — APPROVED** [review]
Thanks for noticing this and fixing it!

File: src/Core/Azure.DataApiBuilder.Core.csproj (hunk 1 above)
Comment (Aniruddh25) [reply src/Core/Azure.DataApiBuilder.Core.csproj:7]: I dont think they mean actual build executable but other source code files which could be ignored by source control yet be linked into the PDB, perhaps could also mean HotChocolate dependencies which are imported from a source package

**Pull request merged**